domain/bakerystorage/state/config.go

```diff
@@ -0,0 +1,142 @@
+	offersThirdPartyKey *bakery.KeyPair,
+) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	cfg := Config{
+		LocalUsersPrivateKey:              &keyScanner{localUsersKey.Private.Key},
+		LocalUsersPublicKey:               &keyScanner{localUsersKey.Public.Key},
+		LocalUsersThirdPartyPrivateKey:    &keyScanner{localUsersThirdPartyKey.Private.Key},
+		LocalUsersThirdPartyPublicKey:     &keyScanner{localUsersThirdPartyKey.Public.Key},
+		ExternalUsersThirdPartyPrivateKey: &keyScanner{externalUsersThirdPartyKey.Private.Key},
+		ExternalUsersThirdPartyPublicKey:  &keyScanner{externalUsersThirdPartyKey.Public.Key},
+		OffersThirdPartyPrivateKey:        &keyScanner{offersThirdPartyKey.Private.Key},
+		OffersThirdPartyPublicKey:         &keyScanner{offersThirdPartyKey.Public.Key},
+	}
+
+	initialiseConfigStmt, err := sqlair.Prepare("INSERT INTO bakery_config (*) VALUES ($Config.*)", Config{})
+	if err != nil {
+		return errors.Trace(err)
+	}
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, initialiseConfigStmt, cfg).Run()
+		if internaldatabase.IsErrConstraintUnique(err) {
+			return BakeryConfigAlreadyInitialised
+		}
+		return domain.CoerceError(err)
+	})
+	return errors.Trace(err)
+}
+
+func (st *ConfigState) GetLocalUsersKey(ctx context.Context) (*bakery.KeyPair, error) {
+	cfg, err := st.getConfig(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return &bakery.KeyPair{
+		Public:  bakery.PublicKey{Key: cfg.LocalUsersPublicKey.Key()},
+		Private: bakery.PrivateKey{Key: cfg.LocalUsersPrivateKey.Key()},
+	}, nil
+}
+
+func (st *ConfigState) GetLocalUsersThirdPartyKey(ctx context.Context) (*bakery.KeyPair, error) {
+	cfg, err := st.getConfig(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return &bakery.KeyPair{
+		Public:  bakery.PublicKey{Key: cfg.LocalUsersThirdPartyPublicKey.Key()},
+		Private: bakery.PrivateKey{Key: cfg.LocalUsersThirdPartyPrivateKey.Key()},
+	}, nil
+}
+
+func (st *ConfigState) GetExternalUsersThirdPartyKey(ctx context.Context) (*bakery.KeyPair, error) {
+	cfg, err := st.getConfig(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return &bakery.KeyPair{
+		Public:  bakery.PublicKey{Key: cfg.ExternalUsersThirdPartyPublicKey.Key()},
+		Private: bakery.PrivateKey{Key: cfg.ExternalUsersThirdPartyPrivateKey.Key()},
+	}, nil
+}
+
+func (st *ConfigState) GetOffersThirdPartyKey(ctx context.Context) (*bakery.KeyPair, error) {
+	cfg, err := st.getConfig(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return &bakery.KeyPair{
+		Public:  bakery.PublicKey{Key: cfg.OffersThirdPartyPublicKey.Key()},
+		Private: bakery.PrivateKey{Key: cfg.OffersThirdPartyPrivateKey.Key()},
+	}, nil
+}
+
+func (st *ConfigState) getConfig(ctx context.Context) (Config, error) {
+	db, err := st.DB()
+	if err != nil {
+		return Config{}, errors.Trace(err)
+	}
+
+	q := "SELECT &Config.* FROM bakery_config"
```

Comment:
**todo:** Don't introduce a variable unless it's necessary it burden the brain for no reason. I think it's better to inline the query here

Reply:
Yeah agree.

In other domain state packages, it is quite common to separate the query string from the call to prepare with a single use var.

However, now that I've looked a little more, I think this is because queries tend to be multi-line